transports/wasm-ext/src/webtransport/error.rs

```diff
@@ -0,0 +1,38 @@
+use libp2p_core::transport::TransportError;
+use wasm_bindgen::{JsCast, JsValue};
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Listening on WebTransport is not possible from within a browser")]
+    ListeningNotPossible,
+
+    #[error("Invalid multiaddr: {0}")]
+    InvalidMultiaddr(&'static str),
+
+    #[error("Noise error: {0}")]
+    Noise(#[from] libp2p_noise::Error),
```

Comment:
This will duplicate the error string of noise because it is also set as `source`.


```suggestion
    #[error("Noise authentication failed")]
    Noise(#[from] libp2p_noise::Error),
```